### PR TITLE
Utils enhancements to allow go support for JFrog Curation

### PR DESCRIPTION
**Pull request by asafambar**
- [X] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [X] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [X] This pull request is on the dev branch.
- [X] I used gofmt for formatting the code before submitting the pull request.
-----

Depends on: https://github.com/jfrog/build-info-go/pull/255

### CODE REVIEW

**Review by attiasas — COMMENTED** [review]
* Please add test cases for the new bool argument of `getArtifactoryApiUrl`

File: artifactory/commands/golang/go.go [hunk 1]
```diff
@@ -330,19 +330,25 @@ func buildPackageVersionRequest(name, branchName string) string {
...
-	repoUrl += "|direct"
+	if !isCurationCmd {
+		repoUrl += "|direct"
+	}
 	return os.Setenv("GOPROXY", repoUrl)
 }
+
+func SetGoModCache(cacheFolder string) error {
+	return os.Setenv("GOMODCACHE", cacheFolder)
+}
```
Comment (attiasas) [inline artifactory/commands/golang/go.go:354]:
```suggestion
const GoModCacheEnvVar = "GOMODCACHE"
func SectGoModCache(cacheFolder string) (err error, restore func() error) {
	oldVal, exists := os.LookupEnv(GoModCacheEnvVar)
	return errorutils.CheckError(os.Setenv(GoModCacheEnvVar, cacheFolder)), func() error {
		if !exists {
			return errorutils.CheckError(os.Unsetenv(GoModCacheEnvVar))
		} else {
			return errorutils.CheckError(os.Setenv(GoModCacheEnvVar, oldVal))
		}
	}
}
```
You can add option to restore env when done, most of the times when we end our commands we want to restore the user env

Comment (asafambar) [reply artifactory/commands/golang/go.go:354]: it shouldnt change the user env, only in this session maybe, but I'm not sure

File: utils/golang/utils.go [hunk 2]
```diff
@@ -83,8 +84,8 @@ func GetModuleName(projectDir string) (string, error) {
 	return path, nil
 }
 
-func GetDependenciesList(projectDir string) (map[string]bool, error) {
-	deps, err := utils.GetDependenciesList(projectDir, log.Logger)
+func GetDependenciesList(projectDir string, errorFunc utils.HandleErrorFunc) (map[string]bool, error) {
```
Comment (eyalbe4) [inline utils/golang/utils.go:87]: This function seems to be unused, so let's remove it.
The same goes for the below `GetDependenciesGraph` function. Let'e remove it as well.

File: utils/golang/utils.go [hunk 3]
```diff
@@ -99,17 +100,41 @@ func GetDependenciesGraph(projectDir string) (map[string][]string, error) {
...
 		return "", err
 	}
-	return getArtifactoryApiUrl(repo, authServerDetails)
+	return getArtifactoryApiUrl(repo, authServerDetails, goProxyParams)
 }
 
 // Gets the URL of the specified repository Go API in Artifactory.
 // The URL contains credentials (username and access token or password).
-func getArtifactoryApiUrl(repoName string, details auth.ServiceDetails) (string, error) {
+func getArtifactoryApiUrl(repoName string, details auth.ServiceDetails, goProxyParams GoProxyUrlParams) (string, error) {
```
Comment (eyalbe4) [inline utils/golang/utils.go:137]: You enhanced this function by adding a new `goProxyParams` argument to it, however, you did not enhance the existing `TestGetArtifactoryApiUrl` function to also test new argument functionality.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
See my inline comments.
I'd love to run another review cycle following the implementation of my comments.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
Let's also create a PR for github.com/jfrog/jfrog-cli with an upgrade of the jfrog-cli-core and build-info-go modules that include the changes proposed in their PRs, and ensure the tests there are passing.
Let's link that jfrog-cli PR to this PR and the build-info-go PR.

File: utils/golang/utils.go (hunk 2 above)
Comment (asafambar) [reply utils/golang/utils.go:87]: It's in use in the `security` repo

**Comment by asafambar** [comment]
> Let's also create a PR for github.com/jfrog/jfrog-cli with an upgrade of the jfrog-cli-core and build-info-go modules that include the changes proposed in their PRs, and ensure the tests there are passing. Let's link that jfrog-cli PR to this PR and the build-info-go PR.

@attiasas  will take care of that

File: utils/golang/utils.go (hunk 2 above)
Comment (eyalbe4) [reply utils/golang/utils.go:88]: If those functions are used by the jfrog-cli-security module, let's move them to that module.

Comment (asafambar) [reply utils/golang/utils.go]: Done

**Pull request merged**